### PR TITLE
Linux: Fix build with `use_sowrap=no` and various warnings/errors

### PR DESCRIPTION
**Pull request by akien-mga**
Follow-up to #74978 and #76961.
Still testing it, some more changes might be needed.

Fixes this error and these warnings:
```
platform/linuxbsd/x11/display_server_x11.cpp: In constructor 'DisplayServerX11::DisplayServerX11(const String&, DisplayServer::WindowMode, DisplayServer::VSyncMode, uint32_t, const Vector2i*, const Vector2i&, int, Error&)':
platform/linuxbsd/x11/display_server_x11.cpp:5452:9: error: 'xkb_loaded' was not declared in this scope
 5452 |         xkb_loaded = true;
      |         ^~~~~~~~~~
platform/linuxbsd/x11/display_server_x11.cpp:5480:22: warning: the address of 'XcursorImage* XcursorImageCreate(int, int)' will never be NULL [-Waddress]
 5480 |                 if (!XcursorImageCreate || !XcursorImageLoadCursor || !XcursorImageDestroy || !XcursorGetDefaultSize || !XcursorGetTheme || !XcursorLibraryLoadImage) {
      |                      ^~~~~~~~~~~~~~~~~~
In file included from platform/linuxbsd/x11/display_server_x11.h:91,
                 from platform/linuxbsd/x11/display_server_x11.cpp:31:
/usr/include/X11/Xcursor/Xcursor.h:237:1: note: 'XcursorImage* XcursorImageCreate(int, int)' declared here
  237 | XcursorImageCreate (int width, int height);
      | ^~~~~~~~~~~~~~~~~~
platform/linuxbsd/x11/display_server_x11.cpp:5480:45: warning: the address of 'Cursor XcursorImageLoadCursor(Display*, const XcursorImage*)' will never be NULL [-Waddress]
 5480 |                 if (!XcursorImageCreate || !XcursorImageLoadCursor || !XcursorImageDestroy || !XcursorGetDefaultSize || !XcursorGetTheme || !XcursorLibraryLoadImage) {
      |                                             ^~~~~~~~~~~~~~~~~~~~~~
/usr/include/X11/Xcursor/Xcursor.h:386:1: note: 'Cursor XcursorImageLoadCursor(Display*, const XcursorImage*)' declared here
  386 | XcursorImageLoadCursor (Display *dpy, const XcursorImage *image);
      | ^~~~~~~~~~~~~~~~~~~~~~
platform/linuxbsd/x11/display_server_x11.cpp:5480:72: warning: the address of 'void XcursorImageDestroy(XcursorImage*)' will never be NULL [-Waddress]
 5480 |                 if (!XcursorImageCreate || !XcursorImageLoadCursor || !XcursorImageDestroy || !XcursorGetDefaultSize || !XcursorGetTheme || !XcursorLibraryLoadImage) {
      |                                                                        ^~~~~~~~~~~~~~~~~~~
/usr/include/X11/Xcursor/Xcursor.h:240:1: note: 'void XcursorImageDestroy(XcursorImage*)' declared here
  240 | XcursorImageDestroy (XcursorImage *image);
      | ^~~~~~~~~~~~~~~~~~~
platform/linuxbsd/x11/display_server_x11.cpp:5480:96: warning: the address of 'int XcursorGetDefaultSize(Display*)' will never be NULL [-Waddress]
 5480 |                 if (!XcursorImageCreate || !XcursorImageLoadCursor || !XcursorImageDestroy || !XcursorGetDefaultSize || !XcursorGetTheme || !XcursorLibraryLoadImage) {
      |                                                                                                ^~~~~~~~~~~~~~~~~~~~~
/usr/include/X11/Xcursor/Xcursor.h:485:1: note: 'int XcursorGetDefaultSize(Display*)' declared here
  485 | XcursorGetDefaultSize (Display *dpy);
      | ^~~~~~~~~~~~~~~~~~~~~
platform/linuxbsd/x11/display_server_x11.cpp:5480:122: warning: the address of 'char* XcursorGetTheme(Display*)' will never be NULL [-Waddress]
 5480 |                 if (!XcursorImageCreate || !XcursorImageLoadCursor || !XcursorImageDestroy || !XcursorGetDefaultSize || !XcursorGetTheme || !XcursorLibraryLoadImage) {
      |                                                                                                                          ^~~~~~~~~~~~~~~
/usr/include/X11/Xcursor/Xcursor.h:491:1: note: 'char* XcursorGetTheme(Display*)' declared here
  491 | XcursorGetTheme (Display *dpy);
      | ^~~~~~~~~~~~~~~
platform/linuxbsd/x11/display_server_x11.cpp:5480:142: warning: the address of 'XcursorImage* XcursorLibraryLoadImage(const char*, const char*, int)' will never be NULL [-Waddress]
 5480 |                 if (!XcursorImageCreate || !XcursorImageLoadCursor || !XcursorImageDestroy || !XcursorGetDefaultSize || !XcursorGetTheme || !XcursorLibraryLoadImage) {
      |                                                                                                                                              ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/X11/Xcursor/Xcursor.h:366:1: note: 'XcursorImage* XcursorLibraryLoadImage(const char*, const char*, int)' declared here
  366 | XcursorLibraryLoadImage (const char *library, const char *theme, int size);
      | ^~~~~~~~~~~~~~~~~~~~~~~
Compiling thirdparty/astcenc/astcenc_c
```

And this warning which exposed a logic bug:
```
platform/linuxbsd/joypad_linux.cpp:80:13: warning: 'bool detect_sandbox()' defined but not used [-Wunused-function]
   80 | static bool detect_sandbox() {
      |             ^~~~~~~~~~~~~~
```

### CODE REVIEW

File: platform/linuxbsd/x11/display_server_x11.cpp [hunk 1]
```diff
@@ -5476,6 +5476,7 @@ DisplayServerX11::DisplayServerX11(const String &p_rendering_driver, WindowMode
 
 	r_error = OK;
 
+#ifdef USE_SOWRAP
```
Comment (bruvzg) [inline platform/linuxbsd/x11/display_server_x11.cpp:5479]:
```suggestion
#ifdef SOWRAP_ENABLED
```

Comment (akien-mga) [reply platform/linuxbsd/x11/display_server_x11.cpp:5479]: Oops, thanks.

**Comment by akien-mga** [comment]
It passed building on the Mageia buildsystem with this so it should be good to review. I'll still do some sanity checks on the generated package before merging.

File: platform/linuxbsd/x11/display_server_x11.cpp [hunk 2]
```diff
@@ -5449,7 +5449,7 @@ DisplayServerX11::DisplayServerX11(const String &p_rendering_driver, WindowMode
 	}
 #else
 #ifdef XKB_ENABLED
-	xkb_loaded = true;
+	bool xkb_loaded = true;
```
Comment (bruvzg) [inline platform/linuxbsd/x11/display_server_x11.cpp:5452]: We probably should set `xkb_loaded_v05p` and `xkb_loaded_v08p` to `true` as well. Current version is 1.5.0, so it's probably reasonable to expect at leas 0.8 when linking system version.

Comment (akien-mga) [reply platform/linuxbsd/x11/display_server_x11.cpp:5452]: Thanks, fixed.

**Comment by YuriSizov** [comment]
Cherry-picked for 4.1.1.

**Pull request merged**